oqpy/control_flow.py

```diff
@@ -132,6 +133,8 @@ def ForIn(
     elif isinstance(iterator, _ClassicalVar):
         set_declaration = to_ast(program, iterator)
         assert isinstance(set_declaration, ast.Identifier), type(set_declaration)
+    elif isinstance(iterator, RangeDefinition):
+        set_declaration = iterator.to_ast(program)
```

Comment:
A few lines above there is already a case where `iterator` is `range` - it calls the existing `convert_range` function which I think does essentially what you are trying to do here. Is there a way to use that codepath instead of creating a new case here?

Reply:
same as [above](https://github.com/ajberdy/oqpy/pull/4#discussion_r1224560485). I can move the code path into `convert_range` but there will still be two code paths.

Reply:
In the current implementation, `RangeDefinition` is a concrete oqpy expression that has `type` and `to_ast` defined